### PR TITLE
jsonpb: check for nil in Marshal and return error to avoid panic.

### PR DESCRIPTION
**Pull request by cybrcodr**
Add check for nil in Marshal and return error to avoid panic.

### CODE REVIEW

File: jsonpb/jsonpb_test.go [hunk 1]
```diff
@@ -462,6 +462,14 @@ func TestMarshaling(t *testing.T) {
 	}
 }
 
+func TestMarshalingNil(t *testing.T) {
+	var msg *pb.Simple
+	m := &Marshaler{}
+	if _, err := m.MarshalToString(msg); err == nil {
+		t.Errorf("mashalling nil returned no error")
```
Comment (dsnet) [inline jsonpb/jsonpb_test.go]: Nit: It seems the convention is usually one l.

s/marshalling/marshaling/

File: jsonpb/jsonpb.go [hunk 2]
```diff
@@ -118,6 +118,9 @@ type JSONPBUnmarshaler interface {
 
 // Marshal marshals a protocol buffer into JSON.
 func (m *Marshaler) Marshal(out io.Writer, pb proto.Message) error {
+	if pb == nil || reflect.ValueOf(pb).IsNil() {
```
Comment (dsnet) [inline jsonpb/jsonpb.go]: This assumes that pb is a pointer, which is not always true.

Comment (cybrcodr) [reply jsonpb/jsonpb.go]: Ah yes.  Switched to checking for pointer only.  I'm guessing I shouldn't worry about nil maps, slices, etc.

File: jsonpb/jsonpb_test.go (hunk 1 above)
Comment (cybrcodr) [reply jsonpb/jsonpb_test.go]: Done.

**Pull request merged**